test/integration/auth_server_test.go

```diff
@@ -0,0 +1,33 @@
+// +build integration
```

Comment:
I do not like the name of that file. IMO it should be placed in suite_test.go or in helpers.go

for me right now it's like in that file I should see the test for the auth_server and not helper for setting up the HTTP server with BasicAuth


What's more, it's that really necessary?
you can directly replace this
```go
repoServer := httptest.NewServer(&authServer{
		authRequired: tsCfg.httpBasicAuth,
		target:       http.FileServer(http.Dir("testdata")),
		password:     basicPassword,
		username:     basicUsername,
		t:            t,
	})
```

with this
```go
httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if tsCfg.httpBasicAuth {
			u, p, ok := r.BasicAuth()
			assert.True(t, ok, "basic auth required")
			assert.Equal(t, basicUsername, u)
			assert.Equal(t, basicPassword, p)
		}

		http.FileServer(http.Dir("testdata")).ServeHTTP(w, r)
	}))
```

benefits? 
- one file and 33 lines of code can be removed, with the same level of readability